pkg/kubelet/server/server.go

```diff
@@ -938,18 +957,18 @@ func (s *Server) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	}
 
 	method, path := s.getMetricMethodBucket(req.Method), s.getMetricBucket(req.URL.Path)
-
 	longRunning := strconv.FormatBool(isLongRunningRequest(path))
 
-	servermetrics.HTTPRequests.WithLabelValues(method, path, serverType, longRunning).Inc()
+	startTime := time.Now()
 
-	servermetrics.HTTPInflightRequests.WithLabelValues(method, path, serverType, longRunning).Inc()
-	defer servermetrics.HTTPInflightRequests.WithLabelValues(method, path, serverType, longRunning).Dec()
+	handler.ServeHTTP(&rw, req)
```

Comment:
Moving ServeHTTP in relation to the metric calls means we are fundamentally changing the behavior in the event of an exception. I think it is better to record the metrics about the incoming problematic request in that case.

Reply:
When record the metrics about the incoming problematic request, Can`t record the http response status code. So i am Moving ServeHTTP in relation to the metric calls.

Reply:
@tallclair, what do you think of this code change

Reply:
We could probably guess the status code if ServeHTTP throws an exception (500) but I agree we cannot be sure.  I'm not sure that not knowing the status code of the response is a good enough reason to not record that we got the call.